tests/main/apparmor-prompting-snapd-startup/task.yaml

```diff
@@ -100,3 +100,67 @@ execute: |
     echo "Check that only the non-expired rule is still valid (must be done with UID 1000)"
     sudo -iu '#1000' snap debug api /v2/interfaces/requests/rules | jq '.result | length' | MATCH 1
     sudo -iu '#1000' snap debug api /v2/interfaces/requests/rules | jq '.result.[0].id' | MATCH "0000000000000002"
+
+    echo '### Simulate failure to open interfaces requests manager ###'
+
+    echo "Stop snapd and ensure it is not in failure mode"
+    systemctl stop snapd.service snapd.socket
+    # Try for a while to make sure it's not in failure mode
+    echo "Check that systemctl is-failed is never true after a while"
+    retry --wait 1 -n 30 systemctl is-failed snapd.service snapd.socket && exit 1
```

Comment:
nitpick: this relies in very hidden behavior of `set -e`, I think that `not retry ...` would work too

Reply:
Thanks, I'll try it with this instead